keps/sig-node/2400-node-swap/README.md

```diff
@@ -274,6 +301,50 @@ In this example, Container A would have a swap limit of 19 GB, and Container B w
 
 This approach allocates swap limits based on each container's memory request and adjusts the proportion based on the total swap memory available in the system. It ensures that each container gets a fair share of the swap space and helps maintain resource allocation efficiency.
 
+### Swap-aware Evictions
+
+As part of this KEP, the eviction manager will be enhanced to be swap-aware.
+This update will enable the eviction manager to account for swap usage in its decision-making process.
+By doing so, it will help prevent the system from exhausting swap space, thereby maintaining system stability and responsiveness.
+
+#### Background
+
+Before this KEP, kubelet's eviction manager completely overlooked swap memory, leading to several issues:
+* Inaccessible Swap: The memory eviction threshold is configured in such a way that swap is never triggered during node-level pressure, 
+as eviction occurs before the node starts swapping memory.
+* Unfairness & Instability: The eviction manager may evict the "wrong" or innocent pods, failing to address the actual memory pressure.
+* Unexpected Behavior: Pods that exceed their memory limits (with regular and swap memory) are not evicted first,
+even though they would immediately get killed if swap were not used.
+
+Here we present an extension to the eviction manager that will address these issues by becoming swap-aware.
+The proposed logic is fully backward compatible and requires no additional configuration, making it completely transparent to the user.
+
+To achieve this, we recommend enhancing the eviction manager's memory pressure handling to account for swap memory, rather than adding a distinct swap signal.
+Memory and swap are inherently connected and should be addressed as a single issue.
+By integrating swap memory into the eviction manager's logic, we ensure a more accurate and efficient handling of system resources.
+For example, separating memory and swap memory is problematic because swap is not used until memory is full.
+
+#### Defining "accessible swap"
+
+Let `accessible swap` be the amount of swap that is accessible by pods according to the [LimitedSwap swap behavior](#steps-to-calculate-swap-limit).
+Note that the amount of accessible swap changes in time according to the pods running on the node.
```

Comment:
I think this could use reference as to why: not every pod is allocated swap, and the full swap space isn't used. In max swap case every pod is BestEffort and is allocated the swap/memory ratio, but even then there (should be) system reserved space that will take away from workload memory. Thus the max amount of available swap to workloads is the  sum of the allocated swap of each best effort pod